### PR TITLE
[CDAP-17098] Add widget for Window Aggregation configuration DO NOT MERGE

### PR DESCRIPTION
**Pull request by njbriggs**
This adds a new widget to support the Window Aggregation plugin. It builds on the widget used by Group By and adds an optional arguments field and an "ignore nulls" checkbox.

image

Also includes unit tests for parsing and serializing the row configuration.

Do not merge until we know if/when this is going into 6.1.x. When it looks good, I'll create PRs for 6.2 and develop.

### CODE REVIEW

**Comment by nitinmotgi** [comment]
Nick, anyway we line the “Ignore Nulls” and “as” text with rest of the text
in that row ?

**Comment by njbriggs** [comment]
> Nick, anyway we line the “Ignore Nulls” and “as” text with rest of the text in that row ?

@nitinmotgi I will look at the vertical alignment.

File: cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownArgumentsWidget/FunctionDropdownArgumentsRow.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+const styles = (theme): StyleRules => {
+  return {
+    ...AbstractRowStyles(theme),
+    inputContainer: {
+      display: 'grid',
+      gridTemplateColumns: '1fr 1fr 1fr 1fr 30px 1fr',
+      gridGap: '10px',
+    },
+    disabled: {
+      color: `${theme.palette.grey['50']}`,
```
Comment (jennac3) [inline cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownArgumentsWidget/FunctionDropdownArgumentsRow.tsx:42]: We don't need to wrap it with `${}`, I believe we can do `color: theme.palette.grey['50']`

**Comment by njbriggs** [comment]
Updated to address visual comments:
image

**Comment by ajainarayanan** [comment]
> Updated to address visual comments:
> image

I think this will eventually come, but the input field selector without border bottom seems off in that row. The dropdown to the left seems to have the border bottom. Is this a conscious choice?

**Comment by njbriggs** [comment]
> I think this will eventually come, but the input field selector without border bottom seems off in that row. The dropdown to the left seems to have the border bottom. Is this a conscious choice?

I think that's caused by how `InputDropdownField` uses `InputBase`. It looks like `InputBase` doesn't add the underline.

**Comment by njbriggs** [comment]
Updated to use only `FormInputs`:
image

**Comment by nitinmotgi** [comment]
Thank you Nick, this looks much better.

**Comment by ajainarayanan** [comment]
> Updated to use only `FormInputs`:
> image

One request. Can we add the border bottom to all of them? Without the borders now there is no alignment per line. All the input elements in multi-row widget has a border 
Screen Shot 2020-08-06 at 10 25 34 PM

File: cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownArgumentsWidget/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+export default function FunctionDropdownArgumentsWidget(props) {
+  return (
+    <ThemeWrapper>
+      <FunctionDropdownArgumentsWidgetView {...props} />
+    </ThemeWrapper>
+  );
+}
+
+(FunctionDropdownArgumentsWidget as any).propTypes = WIDGET_PROPTYPES;
```
Comment (jennac3) [inline cdap-ui/app/cdap/components/AbstractWidget/FunctionDropdownArgumentsWidget/index.tsx:70]: I was wondering, if this widget will be supported in Plugin JSON Creator, should we add `getWidgetAttributes` as following?
https://github.com/cdapio/cdap/blob/b409da5b407dc817d8facad606c3928b6ca9f4a0/cdap-ui/app/cdap/components/AbstractWidget/FormInputs/Select/index.tsx#L166

**Comment by njbriggs** [comment]
> Can we add the border bottom to all of them?

Let's talk about this tomorrow. The FormInputs use `InputBase` from material-ui which has no underline styling. The underline styling is added by `Input` which is used in some of the other multi-row widgets (like `KeyValueWidget`, `FunctionDropdownAlias`, etc.).

**Comment by njbriggs** [comment]
Replaced by #12708

**Pull request closed without merging**